src/overlays/actors/ovl_Bg_Mizu_Water/z_bg_mizu_water.c

```diff
@@ -21,19 +43,305 @@ const ActorInit Bg_Mizu_Water_InitVars = {
+    u32 pad;
+    s16 params;
+    s32 waterLevelActionIndex;
+    WaterBox* waterBoxes;
+
+    waterBoxes = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    switch (this->type) {
+        case 0:
+            params = this->actor.params;
+            waterLevelActionIndex = BgMizuWater_getWaterLevelActionIndex(this->actor.params, globalCtx);
+            if (waterLevelActionIndex != 0) {
+                if (params != waterLevels[waterLevelActionIndex].switchID) {
+                    this->actor.params = waterLevels[waterLevelActionIndex].switchID;
+                    this->targetY = waterLevels[waterLevelActionIndex].yDiff + this->baseY;
+                }
+            }
+
+            if ((params != this->actor.params) && (params != 0))
+                Flags_UnsetSwitch(globalCtx, params);
+
+            if (Math_ApproxF(&this->actor.posRot.pos.y, this->targetY, 5.0f) != 0) {
+                globalCtx->unk_11D30[0] = 0;
+                this->actionFunc = BgMizuWater_Idle;
+                func_80106CCC(globalCtx);
+            }
+            BgMizuWater_setWaterBoxesHeights(globalCtx->colCtx.stat.colHeader->waterBoxes, this->actor.posRot.pos.y);
+            break;
+        case 1:
+            break;
+        case 2:
+            if (Flags_GetSwitch(globalCtx, this->switchTriggerID)) {
+                this->targetY = this->baseY + 85.0f;
+            } else {
+                this->targetY = this->baseY;
+            }
+            if (Math_ApproxF(&this->actor.posRot.pos.y, this->targetY, 1.0f) != 0) {
+                globalCtx->unk_11D30[0] = 0;
+                this->actionFunc = BgMizuWater_Idle;
+            }
+            waterBoxes[6].unk_02 = this->actor.posRot.pos.y;
+            break;
+        case 3:
+            if (Flags_GetSwitch(globalCtx, this->switchTriggerID)) {
+                this->targetY = this->baseY + 110.0f;
+            } else {
+                this->targetY = this->baseY;
+            }
+            if (Math_ApproxF(&this->actor.posRot.pos.y, this->targetY, 1.0f) != 0) {
+                globalCtx->unk_11D30[0] = 0;
+                this->actionFunc = BgMizuWater_Idle;
+            }
+            waterBoxes[8].unk_02 = this->actor.posRot.pos.y;
+            break;
+        case 4:
+            if (Flags_GetSwitch(globalCtx, this->switchTriggerID)) {
+                this->targetY = this->baseY + 160.0f;
+            } else {
+                this->targetY = this->baseY;
+            }
+            if (Math_ApproxF(&this->actor.posRot.pos.y, this->targetY, 1.0f) != 0) {
+                globalCtx->unk_11D30[0] = 0;
+                this->actionFunc = BgMizuWater_Idle;
+            }
+            waterBoxes[16].unk_02 = this->actor.posRot.pos.y;
+            break;
+    }
+
+    if (this->targetY < this->actor.posRot.pos.y) {
+        func_800AA000(0.0f, 0x78, 0x14, 0xA);
+        func_8002F948(&this->actor, NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
```

Comment:
`& ~SFX_FLAG` makes more sense than `- SFX_FLAG` for a flag

Reply:
we use - sfx flag for now. can talk about it more in discord if you like, but this is the convention for the time being